scripts/get-badges.ts

```diff
@@ -9,61 +19,116 @@ const usage = () => {
 ---------------------------------------------------------
 ${scriptName}: Get user details. Prints each user's email address, last seen time in milliseconds since epoch, and codes used.
 
-Usage: node ${scriptName} PROJECT_ID VENUE_ID CREDENTIAL_PATH
+Usage: ${scriptName} PROJECT_ID VENUE_IDS [CREDENTIAL_PATH]
 
-Example: node ${scriptName} co-reality-map myawesomevenue prodAccountKey.json
+Example: ${scriptName} co-reality-map venueId,venueId2,venueIdN [someAccountKey.json]
 ---------------------------------------------------------
 `;
 
   console.log(helpText);
   process.exit(1);
 };
 
-const [projectId, venueId] = process.argv.slice(2);
-if (!projectId || !venueId) {
+const [projectId, venueIds, credentialPath] = process.argv.slice(2);
+
+// Note: no need to check credentialPath here as initFirebaseAdmin defaults it when undefined
+if (!projectId || !venueIds) {
   usage();
 }
 
-const app = initFirebaseAdminApp(projectId, {
-  appName: projectId,
+const venueIdsArray = venueIds.split(",");
+
+initFirebaseAdminApp(projectId, {
+  credentialPath: credentialPath
+    ? resolve(__dirname, credentialPath)
+    : undefined,
 });
 
+interface UsersWithVisitsResult {
+  user: WithId<User>;
+  visits: WithId<UserVisit>[];
+}
+
 (async () => {
-  const allUsers: admin.auth.UserRecord[] = [];
-  let nextPageToken: string | undefined;
-  const { users, pageToken } = await app.auth().listUsers(1000);
+  // TODO: extract this as a generic helper function?
+  const usersWithVisits = await Promise.all(
+    await admin
+      .firestore()
+      .collection("users")
+      .where("enteredVenueIds", "array-contains-any", venueIdsArray)
```

Comment:
Is it possible to have `venueIdsArray.length > 10`?

Reply:
As the code is written, technically yes.

I'm not sure if we would ever need to do so in practice though, possibly for a large event if we wanted to check all 'related venues' specifically (and not just use the 'parent venue' as a 'gateway'), before https://github.com/sparkletown/internal-sparkle-issues/issues/127 is implemented.

If it ends up being not too hard of a code change, might be worth implementing the 'chunk' pattern here. If it turns out to be a lot of extra work, probably just check that `venueIdsArray.length <= 10` at the start of the script, and throw an error if not.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1065/commits/6dd0b317b5abd0a38dd40905ecf2458904b25649

Reply:
Please, don't resolve comments, since it is hard to keep up with all the changes with them resolved

Reply:
Does the comment saying that it was fixed and where not solve that need?

I know this is the same thing I’ve requested of you in the past, so it’s a reasonable request. I probably resolved it by habit because I knew it was done + referenced where/why, so by doing so it removed it from the things I need to mentally process in ensuring the PR is ready.

I think where I find it ‘harder’ (and likely the original source of my request) is if someone just ‘replies’ with a ‘reaction’ (eg. thumbs up) and no comment, then resolves it; as that doesn’t trigger any of the notifications that I often will use to ensure I have checked it all.

Reply:
(referenced this thread in https://github.com/sparkletown/sparkle/pull/1062#issuecomment-764969428 so that we can document it once we figure out our patterns here)